packages/@jsii/go-runtime/jsii-runtime-go/cast.go

```diff
@@ -0,0 +1,53 @@
+package jsii
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/aws/jsii-runtime-go/internal/kernel"
+)
+
+// UnsafeCast converts the given interface value to the desired target interface
+// pointer. Panics if the from value is not a jsii proxy object, or if the to
+// value is not a pointer to an interface type.
+func UnsafeCast(from interface{}, into interface{}) {
+	rinto := reflect.ValueOf(into)
+	if rinto.Kind() != reflect.Ptr {
+		panic(fmt.Errorf("Second argument to UnsafeCast must be a pointer to an interface. Received %s", rinto.Type().String()))
+	}
+	rinto = rinto.Elem()
+	if rinto.Kind() != reflect.Interface {
+		panic(fmt.Errorf("Second argument to UnsafeCast must be a pointer to an interface. Received pointer to %s", rinto.Type().String()))
+	}
+
+	rfrom := reflect.ValueOf(from)
+
+	// If rfrom is essentially nil, set into to nil and return.
+	if !rfrom.IsValid() || rfrom.IsZero() {
+		null := reflect.Zero(rinto.Type())
+		rinto.Set(null)
+		return
+	}
+	// Interfaces may present as a pointer to an implementing struct, and that's fine...
+	if rfrom.Kind() != reflect.Interface && rfrom.Kind() != reflect.Ptr {
+		panic(fmt.Errorf("First argument to UnsafeCast must be an interface value. Received %s", rfrom.Type().String()))
+	}
+
+	// If rfrom can be directly converted to rinto, just do it.
+	if rfrom.CanConvert(rinto.Type()) {
```

Comment:
Looks like `CanConvert` is a new addition in v1.17. I'm fine with bumping JSII minimum if this is a significant convencience (since we are considering a bump to 1.18 for generics anyway)